### PR TITLE
Parser: use C saxpy/sgemm provided by the Ops implementation

### PR DESCRIPTION
**Pull request by danieldk**
## Description

This is a backport of https://github.com/explosion/spaCy/pull/10747 from the parser refactor branch. It eliminates the explicit calls to BLIS, instead using the saxpy/sgemm provided by the Ops implementation.

This allows us to use Accelerate in the parser on M1 Macs (with an updated thinc-apple-ops).

Performance of the `de_core_news_lg` pipe:

```
BLIS 0.7.0, no thinc-apple-ops:  6385 WPS
BLIS 0.7.0, thinc-apple-ops:    36455 WPS
BLIS 0.9.0, no thinc-apple-ops: 19188 WPS
BLIS 0.9.0, thinc-apple-ops:    36682 WPS
This PR, thinc-apple-ops:       38726 WPS
```

Performance of the `de_core_news_lg` pipe (only `tok2vec -> parser`):

```
BLIS 0.7.0, no thinc-apple-ops: 13907 WPS
BLIS 0.7.0, thinc-apple-ops:    73172 WPS
BLIS 0.9.0, no thinc-apple-ops: 41576 WPS
BLIS 0.9.0, thinc-apple-ops:    72569 WPS
This PR, thinc-apple-ops:       87061 WPS
```

### Types of change

Enhancement, performance improvement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — APPROVED** [review]

**Comment by danieldk** [comment]
No more draft, this should work now with Richard's typing PR merged.

File: spacy/pipeline/transition_parser.pyx [hunk 1]
```diff
@@ -259,6 +259,12 @@ cdef class Parser(TrainablePipe):
     def greedy_parse(self, docs, drop=0.):
         cdef vector[StateC*] states
         cdef StateClass state
+        ops = self.model.ops
+        cdef CBlas cblas
+        if isinstance(ops, CupyOps):
+            cblas = get_ops("cpu").cblas()
+        else:
+            cblas = self.model.ops.cblas()
```
Comment (shadeMe) [inline spacy/pipeline/transition_parser.pyx:267]:
```suggestion
            cblas = ops.cblas()
```

**Review by shadeMe — APPROVED** [review]
LGTM with just one minor nitpick.

**Comment by danieldk** [comment]
@explosion-bot please test_gpu

**Pull request merged**